web/pdf_find_controller.js

```diff
@@ -202,8 +208,22 @@ function normalize(text) {
       }
 
       if (p2) {
-        const hasTrailingDashEOL = p2.endsWith("\n");
-        const len = hasTrailingDashEOL ? p2.length - 2 : p2.length;
+        // Use the NFKC representation to normalize the char.
+        let replacement = CHARACTERS_TO_NORMALIZE[p2];
+        if (!replacement) {
+          replacement = CHARACTERS_TO_NORMALIZE[p2] = p2.normalize("NFKC");
+        }
```

Comment:
By adding to the existing `CHARACTERS_TO_NORMALIZE` object you may end up *subtly changing* the logic if a PDF document contains some pages with, and some pages without, syllables since we create two different regular expressions in that case.

I'd suggest adding e.g.
```javascript
const NFKC_CHARS_TO_NORMALIZE = {};
```
after line https://github.com/mozilla/pdf.js/blob/3078e2c1d9c3d1e6bf79709f2680dfca6b35cc9d/web/pdf_find_controller.js#L48
and then also simplify this code as follows:
```suggestion
        const replacement = NFKC_CHARS_TO_NORMALIZE[p2] ||= p2.normalize("NFKC");
```